meshery

```diff
@@ -127,42 +127,77 @@ unzip ${PWD}/meshery.zip -d ${PWD}/$temporary_dir_name
 install_dir=${MESHERY_INSTALL_DIR}
 if [ -z "$install_dir" ]
 then
-    install_dir="/usr/local/bin"
+install_dir="/usr/local/bin"
 fi
 
 printf "\nInstalling mesheryctl in $install_dir.\n"
 WHOAMI=$(whoami)
 if mv ${PWD}/$temporary_dir_name/mesheryctl $install_dir/mesheryctl ; then
-  echo "mesheryctl installed."
+echo "mesheryctl installed."
 else
-  if sudo mv ${PWD}/$temporary_dir_name/mesheryctl $install_dir/mesheryctl ; then
+if sudo mv ${PWD}/$temporary_dir_name/mesheryctl $install_dir/mesheryctl ; then
     echo "Permission Resolved: mesheryctl installed using sudo permissions."
-  else
+else
     echo "Cannot install mesheryctl. Check permissions of $WHOAMI for $install_dir."
     exit 1
-  fi
+fi
 fi
 
 #Transfering permissions and ownership to client USER
 if [ "$SUDO_USER" == "" ] ;
 then
-	if chown $WHOAMI $install_dir/mesheryctl ; then
+if chown $WHOAMI $install_dir/mesheryctl ; then
     echo "permissions moved to "$WHOAMI
-  else
+else
     echo "Unable to write to $install_dir. Please verify write permission for $WHOAMI."
     exit 1
-  fi
+fi
 else
-	if chown $SUDO_USER /usr/local/bin/mesheryctl ; then
+if chown $SUDO_USER /usr/local/bin/mesheryctl ; then
     echo "permissions moved to "$SUDO_USER
-  else
+else
     echo "Unable to write to $install_dir/mesheryctl. Please verify write permission for $SUDO_USER."
     exit 1
-  fi
+fi
 fi
 
 
 printf "Removing installation files and opening Meshery..."
 rm -rf meshery.zip ${PWD}/$temporary_dir_name/
 
+
+# if ADAPTERS are empty then we'd use default local context else go we'd create custom context, which only fetches required adapters
+# Valid adapter option are
+# cpx
+# consul
+# istio
+# linkerd
+# nsm
+# kuma
+# cpx
+# osm
+# traefik
+
+## Any adapter other than above would be ignored
+if [ ! -z "$ADAPTERS" ]; then
+adpaterString=
+for adapter in $ADAPTERS; do
+    case $adapter in
+        cpx|consul|istio|linkerd|nsm|kuma|cpx|osm)
+            adapterString="$adapterString -a meshery-$adapter "
+        ;;
+        traefik)
+            adapterString="$adapterString -a meshery-$adapter-mesh "
+        ;;
+        *)
+            # We are just neglecting invalid mesh options
+        ;;
+    esac    
+  done
+
+    mesheryctl system context create custom -s $adapterString --yes
```

Comment:
@navendu-pottekkat  **-s** is ensuring that we switched to custom context and **--yes** ensures that all the prompts are satisfied.